### ISSUE TITLE
[IfcConvert] Convert string error

### ISSUE THREAD

**Issue by BlocInBloc** [IfcOpenShell/IfcOpenShell#1016]
Hi, we try to convert ifc to xml which contains special characters in the header (FILE_NAME('éâè!$#'...)) but ge got :
`wstring_convert: to_bytes error`.

[voile.ifc.zip](https://github.com/IfcOpenShell/IfcOpenShell/files/5276642/voile.ifc.zip)

However ifc to dae conversion works.
Would it be possible to manage encoding problems in IfcConvert ?

**Comment by aothms** [comment]
We can probably be more lenient here, but the version of 10303-21 (the .ifc spf step file encoding) we're using strictly forbids characters outside of the ascii 0x20...0x7e range. Later revisions support utf-8, but not IFC. So the file is very much invalid. I'll have a look at where exactly it fails and try to make XML conversion not fail.

**Comment by BlocInBloc** [comment]
I agree ! We have notified the CAD authoring tool which generated this file.
Maybe we should not interrupt the conversion (regardless of the output format) and log a warning ?

As a side note, IFC viewers like BIM Collab or BIM Vision are less strict about it and read the file.

**Comment by BIMvoice** [comment]
This is resolved on the current v0.8 line. The attached voile.ifc has `FILE_NAME('éâè!$#', ...)` with raw non-ASCII author bytes, which is the exact trigger for the old `wstring_convert::to_bytes` error. Converting it to XML now produces a well-formed document (xmllint clean, 1910 element nodes) with no to_bytes error, on both IfcConvert and the current XmlSerializer. Recommend closing as fixed.
